### ISSUE TITLE
Search Results: Move distance below rep name and ensure accessibility

### ISSUE THREAD

**Issue by oddball-lindsay** [department-of-veterans-affairs/va.gov-team#132321]
## Background

Current experience in [Production](https://www.va.gov/get-help-from-accredited-representative/find-rep/):

<img width="776" height="131" alt="Image" src="https://github.com/user-attachments/assets/b5238dc5-7843-4a36-8f9c-51b0d3f8b7e0" />

New experience for v3 ([annotated Figma designs](https://www.figma.com/design/bzbwObT9hiItve0q3cQX9c/ARM-Find-and-Appoint-a-Representative--Veteran-facing-?node-id=17725-18725&t=3Bb15v9hF8K5kQez-4)):

<img width="728" height="95" alt="Image" src="https://github.com/user-attachments/assets/b99bcd42-54f5-4f70-8bab-37efc87f491c" />

### Implementation Notes
1. Visual reorder of distance and rep name
2. [Midpoint Review](https://github.com/department-of-veterans-affairs/va.gov-team/issues/131647) requests for improved accessibility: The distance and provider name should be grouped into a single heading, so that users navigating by headings or viewing the heading structure of the page (common for screen reader users), can find the search result that they are looking for. 
3. Name+distance should be top level heading 

Looking at the implementation in Production, 2 & 3 may already be handled, but worth a double check!

<img width="1008" height="695" alt="Image" src="https://github.com/user-attachments/assets/a4ab31a2-d7ee-41a5-b2a9-33a0453d919f" />

## Acceptance Criteria

- [ ] Visually, the distance is below the representative name
- [ ] To adhere to Midpoint Review accessibility feedback, the distance and provider name should be grouped into a single heading
- [ ] The name and distance combined heading should be a top level heading

**Comment by kristen-brown** [comment]
This is ready for QA on Staging! [Link](https://staging.va.gov/get-help-from-accredited-representative/find-rep/)

A few items to note:
1. As requested for accessibility, I grouped both the rep name and distance under a singular `h3` header. Inside the `h3`, I used a `span` around the distance to style it differently from the rep name.
2. In the design, the text size for the distance info (e.g. "2.78 miles") is somewhere between a standard "h4" and "h5" header size. In the interest of using the standard header styles and not custom styles, I used the slightly larger "h4" style. Therefore, the distance appears slightly larger than in the designs. (I also tried "h5," but it caused the distance to not stand out enough among the rest of the card content.

Screenshots from my accessibility testing (using VoiceOver):

**VoiceOver Header List**
<img width="1905" height="1638" alt="VoiceOver Header List" src="https://github.com/user-attachments/assets/559e9185-5e78-470f-8654-33e9ca8da98c" />

**VoiceOver Focus After Hitting 'Enter'**
<img width="2335" height="1465" alt="VoiceOver Header Focus" src="https://github.com/user-attachments/assets/5a8404ea-33a0-4357-a3c6-dd100a42f813" />
